src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.P2P
+{
+    /// <summary>
+    /// Async loop that discovers new peers to connect to.
+    /// </summary>
+    public sealed class PeerDiscoveryLoop : IDisposable
+    {
+        /// <summary> The async loop we need to wait upon before we can shut down this connector.</summary>
+        private IAsyncLoop asyncLoop;
+
+        /// <summary> Factory for creating background async loop tasks.</summary>
+        private readonly IAsyncLoopFactory asyncLoopFactory;
+
+        /// <summary>Global application life cycle control - triggers when application shuts down.</summary>
+        private readonly INodeLifetime nodeLifetime;
+
+        private readonly NodeConnectionParameters parameters;
+
+        /// <summary> Peer address manager instance, see <see cref="IPeerAddressManager"/>.</summary>
+        private readonly IPeerAddressManager peerAddressManager;
+
+        private readonly int peersToFind;
```

Comment:
This is a new class to lets add comments.

Reply:
Added 👍